mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -696,3 +702,64 @@ def update_artifacts_location(self, run_id, new_artifacts_location):
         with self.ManagedSessionMaker() as session:
             run = session.query(SqlRun).filter_by(run_uuid=run_id).first()
             run.artifact_uri = new_artifacts_location
+
+
+def _get_attributes_filtering_clauses(parsed):
+    clauses = []
+    for sql_statement in parsed:
+        key_type = sql_statement.get('type')
+        key_name = sql_statement.get('key')
+        value = sql_statement.get('value')
+        comparator = sql_statement.get('comparator')
+        if SearchUtils.is_attribute(key_type, comparator):
+            op = SearchUtils.filter_ops.get(comparator)
+            if op:
+                clauses.append(op(getattr(SqlRun, key_name), value))
+    return clauses
+
+
+def _to_sqlalchemy_filtering_statement(sql_statement, session):
+    key_type = sql_statement.get('type')
+    key_name = sql_statement.get('key')
+    value = sql_statement.get('value')
+    comparator = sql_statement.get('comparator')
+
+    if SearchUtils.is_metric(key_type, comparator):
+        entity = SqlLatestMetric
+        entity_key = SqlLatestMetric.key
```

Comment:
Nit: `entity_key` and `entity_value` probably don't need to be defined as new variables in each of these branches since `entity_key = entity.key` and `entity_value = entity.value` in all three cases.

The SQLAlchemy query can be constructed as:
```
        return (
            session
            .query(entity)
            .filter(entity.key == key_name, op(entity.value, value))
            .subquery()
        )
```